include/circt/Dialect/FIRRTL/FIRRTLOps.h

```diff
@@ -174,6 +175,99 @@ struct FirMemory {
   }
   StringAttr getFirMemoryName() const;
 };
+
+// Record of the inner sym name, the module name and the corresponding
+// operation. Also the port index, if the symbol is on a module port.
+struct InnerRefRecord {
+  mlir::StringAttr mod, innerSym;
+  mlir::Operation *op = nullptr;
+  unsigned portIdx = 0;
+  InnerRefRecord(StringAttr mod, StringAttr innerSym, Operation *op)
+      : mod(mod), innerSym(innerSym), op(op) {}
+  InnerRefRecord(StringAttr mod, StringAttr innerSym, Operation *op,
+                 unsigned portIdx)
+      : mod(mod), innerSym(innerSym), op(op), portIdx(portIdx) {}
+  InnerRefRecord(hw::InnerRefAttr ref)
+      : mod(ref.getModule()), innerSym(ref.getName()) {}
+  bool operator<(const InnerRefRecord &rhs) const {
+    return (innerSym.getValue() < rhs.innerSym.getValue() ||
+            (innerSym == rhs.innerSym && mod.getValue() < rhs.mod));
```

Comment:
what about op and portIdx?

Reply:
We want to construct an `InnerRefRecord` with just the module name and the sym name. So, the op and portIdx, might not be available or might not exist.
For example, an invalid `InnerRefAttr` would not have the corresponding op.
So, the `InnerRefRecord` can be constructed with just the module and symbol name, and used for lookups, without considering the op.